cli/azd/cmd/cobra_builder.go

```diff
@@ -126,6 +127,16 @@ func (cb *CobraBuilder) configureActionResolver(cmd *cobra.Command, descriptor *
 				)
 			}
 
+			var suggestionErr *azcli.ErrorWithSuggestion
+			if errors.As(err, &suggestionErr) {
+				cmd.SilenceErrors = true
+				cb.container.Invoke(func(console input.Console) {
+					console.Message(ctx, color.RedString("ERROR: %s", err.Error()))
+					console.Message(ctx, (*azcli.ErrorWithSuggestion)(suggestionErr).Suggestion)
+				})
+				return err
+			}
```

Comment:
@wbreza -- `ActionResult` doesn't fit here because we're determining the cloud configuration before executing actions. `ErrorWithSuggestion` works and this formats all encountered `ErrorWithSuggestion`s accordingly: 

![image](https://github.com/Azure/azure-dev/assets/2158838/cbd67b25-d77f-4e53-930b-b6323ecc410b)

Reply:
@Austinauth right... need to add you to this discussion :)

Reply:
Message and suggestion look great @danieljurek. The suggestion is clear and actionable, and the execution is visually consistent.